state-chain/chains/src/sol/api.rs

```diff
@@ -251,6 +253,30 @@ impl<Environment: SolanaEnvironment> SolanaApi<Environment> {
 		message: Vec<u8>,
 		cf_parameters: Vec<u8>,
 	) -> Result<Self, SolanaTransactionBuildingError> {
+		// Verify the validity of the CCM message before building the call.
+		CcmValidityChecker::<Environment>::is_valid(
+			&CcmChannelMetadata {
+				message: message
+					.clone()
+					.try_into()
+					.expect("This is parsed from bounded vec, therefore the size must fit"),
+				gas_budget: 0,
```

Comment:
Having gas_budget set to zero here looks very strange. I understand that this is because it's very convenient to have the `CcmChannelMetadata` be the type passed to `is_valid`. And then my understanding is that we don't pass the gas_budget we have here because that is different from the `gas_budget` from when opening the deposit channel. If that's true, that's fine for me but maybe let's add a comment explaining it.

Reply:
Added a comment that clarified that the gas_budget is not used by Solana